src/ContainerCommitModal.jsx

```diff
@@ -0,0 +1,191 @@
+            command: this.props.containerWillCommit.Config ? this.props.containerWillCommit.Config.Cmd.join(" ") : "",
+            pause: true,
+            setonbuild: false,
+            onbuild: [""],
+            format: "oci",
+            selectedFormat: "oci",
+            onbuildDisabled: true,
+        };
+        this.state = {...this.initialState};
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.handleCancel = this.handleCancel.bind(this);
+        this.handleCommit = this.handleCommit.bind(this);
+        this.handleOnBuildsInputChange = this.handleOnBuildsInputChange.bind(this);
+        this.handleAddOnBuild = this.handleAddOnBuild.bind(this);
+        this.handleRemoveOnBuild = this.handleRemoveOnBuild.bind(this);
+    }
+
+    handleOnBuildsInputChange(idx, evt) {
+        const newOnbuilds = this.state.onbuild.map((bud, sidx) => {
+            if (idx !== sidx) return bud;
+            console.log(bud);
+            console.log(evt.target.value);
+            bud = evt.target.value;
+            return bud;
+        });
+
+        this.setState({onbuild: newOnbuilds});
+    }
+
+    handleAddOnBuild() {
+        this.setState({onbuild: this.state.onbuild.concat([""])});
+    }
+
+    handleRemoveOnBuild(idx) {
+        this.setState({onbuild: this.state.onbuild.filter((bud, sidx) => idx !== sidx)});
+    }
+
+    handleInputChange(event) {
+        const target = event.target;
+        const value = target.type === 'checkbox' ? target.checked : target.value;
+        const name = target.name;
+        this.setState({
+            [name]: value
+        });
+    }
+
+    handleFormatChange(event) {
+        const selectItem = event.target.value;
+        this.setState({
+            selectedFormat: selectItem,
+            format: selectItem,
+            onbuildDisabled: selectItem === "oci"
+        });
+    }
+
+    // TODO
+    handleCommit() {
+
+    }
+
+    handleCancel() {
+        this.props.handleCancelContainerCommitModal();
+        this.setState(this.initialState);
+    }
+
+    render() {
+        let onbuilds =
+            this.state.onbuild.map((bud, idx) => (
+                <div key={"onbuildvar" + idx} id="select-claimed-onbuildvars" className="containers-run-onbuildvarclaim containers-run-inline" >
+                    <form className="form-inline">
+                        <button type="button" className="btn btn-default fa fa-plus" onClick={this.handleAddOnBuild} />
+                        <button type="button" className="btn btn-default pficon-close" disabled={idx === 0} onClick={() => this.handleRemoveOnBuild(idx)} />
+                        <div className="form-group">
+                            <input type="text" name="onbuildvar_key" className="form-control" onChange={(evt) => this.handleOnBuildsInputChange(idx, evt)} />
+                        </div>
+                    </form>
+                </div>
+            ));
+
+        let commitContent =
```

Comment:
What about implementing the whole modal content with patternfly-react? It will make the code simpler and also get rid of the css.
This modal for example https://rawgit.com/patternfly/patternfly-react/gh-pages/patternfly-3/index.html?selectedKind=patternfly-react%2FForms%20and%20Controls%2FModal%20Overlay&selectedStory=Modal&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

and `show info` to check the code.

I did it recently for some other dialog so I can help you with adjusting it if you need.

Reply:
Thanks, I'll take a look and change it later.

Reply:
Looks much better, thanks!
You can also change the forms to patternfly-react if you want but take that just I a ssuggestion,
Each form group could be something like
``` js
<FormGroup controlId='message' disabled={false}>
      <Grid.Col componentClass={Form.ControlLabel} sm={3}>
        {_("Message")}
      </Grid.Col>
      <Grid.Col sm={9}>
         <input name="message" className="form-control" type="text" onChange={this.handleInputChange} /> 
      </Grid.Col>
</FormGroup>
```